src/lib/string_sign/tests/test_string_sign.ml

```diff
@@ -142,6 +143,36 @@ let test_other_network_failures () =
     "Other network signature fails with different other network" false
     (verify ~signature_kind:(Other_network "Bar") signature keypair.public_key s)
 
+let test_secret_key_between_scalar_field_and_base_field () =
+  (* There are 86663725065984043395317760 values between the two moduli.
+     - Base:
+       28948022309329048855892746252171976963363056481941560715954676764349967630337
+     - Scalar:
+       28948022309329048855892746252171976963363056481941647379679742748393362948097
+  *)
+  let base_modulus =
+    Bignum_bigint.of_string
+      "28948022309329048855892746252171976963363056481941560715954676764349967630337"
+  in
+
+  (* Generate random offset within the range between base and scalar moduli *)
+  let offset_gen = Generator.int64_inclusive 1L Int64.max_value in
+  let random_offset =
+    Quickcheck.random_value offset_gen
+    |> Int64.to_string |> Bignum_bigint.of_string
+  in
+  let sk_bignum = Bignum_bigint.(base_modulus + random_offset) in
+  let sk_str = Bignum_bigint.to_string sk_bignum in
+
+  let secret_key = Signature_lib.Private_key.of_string_exn sk_str in
+  let keypair = Signature_lib.Keypair.of_private_key_exn secret_key in
+
+  let s = "Rain and Spain don't rhyme with cheese" in
+  let signature = sign ~signature_kind:Mainnet keypair.private_key s in
```

Comment:
Usually we use testnet sig kind for all tests. 
If we just don't care sig kind, I'd suggest we generate a sig kind randomly.

Reply:
Updated.